driver/src/main/java/com/code_intelligence/jazzer/driver/FuzzTargetFinder.java

```diff
@@ -0,0 +1,193 @@
+      throw new IllegalArgumentException(String.format(
+          "Methods annotated with @FuzzTest must take a single byte[] or FuzzedDataProvider parameter, got %s for %s in %s",
+          parameter.getName(), method.getName(), clazz.getName()));
+    }
+
+    if (clazz.getDeclaredConstructors().length != 1) {
+      throw new IllegalArgumentException(String.format(
+          "Classes containing a method annotated with @FuzzTest must declare exactly one constructor, got multiple in %s",
+          clazz.getName()));
+    }
+    Constructor<?> constructor = clazz.getDeclaredConstructors()[0];
+    // JUnit 5 supports injected constructor parameters, but we don't.
+    if (constructor.getParameterCount() > 0) {
+      throw new IllegalArgumentException(String.format(
+          "The constructor of a class containing a method annotated with @FuzzTest must take no parameters, got a non-zero number in %s",
+          clazz.getName()));
+    }
+
+    // Both the constructor and the method may not be accessible - JUnit test classes and methods
+    // are usually declared without access modifiers and thus package-private.
+    method.setAccessible(true);
+    constructor.setAccessible(true);
+
+    return Optional.of(new FuzzTarget(parameter == FuzzedDataProvider.class, method,
+        constructor::newInstance, targetPublicStaticMethod(clazz, FUZZER_TEAR_DOWN)));
+  }
+
+  // Finds the traditional static fuzzerTestOneInput fuzz target method.
+  private static FuzzTarget findFuzzTargetByMethodName(Class<?> clazz) {
+    Optional<Method> bytesFuzzTarget =
+        targetPublicStaticMethod(clazz, FUZZER_TEST_ONE_INPUT, byte[].class);
+    Optional<Method> dataFuzzTarget =
+        targetPublicStaticMethod(clazz, FUZZER_TEST_ONE_INPUT, FuzzedDataProvider.class);
+    if (bytesFuzzTarget.isPresent() == dataFuzzTarget.isPresent()) {
+      throw new IllegalArgumentException(String.format(
+          "%s must define exactly one of the following two functions:%n"
+              + "public static void fuzzerTestOneInput(byte[] ...)%n"
+              + "public static void fuzzerTestOneInput(FuzzedDataProvider ...)%n"
+              + "Note: Fuzz targets returning boolean are no longer supported; exceptions should be thrown instead of returning true.",
+          clazz.getName()));
+    }
+
+    Callable<Object> initialize =
+        Stream
+            .of(targetPublicStaticMethod(clazz, FUZZER_INITIALIZE, String[].class)
+                    .map(init -> (Callable<Object>) () -> {
+                      init.invoke(null, (Object) Opt.targetArgs.toArray(new String[] {}));
+                      return null;
+                    }),
+                targetPublicStaticMethod(clazz, FUZZER_INITIALIZE)
+                    .map(init -> (Callable<Object>) () -> {
+                      init.invoke(null);
+                      return null;
+                    }),
+                Optional.of((Callable<Object>) () -> null))
```

Comment:
Remove "Null-Optional" and `getOrElse` as termination operation? Took me some time to understand the IntelliJ warning.

Reply:
Good catch, simplified this a bit, but maybe you see a way to simplify it even further.

Reply:
`Optional`s don't work well with `Stream`s. In other functional languages a `flatMap` should work nicely. I think it's good enough now :smile:

Reply:
That would be possible in Java as well, but both https://docs.oracle.com/javase/9/docs/api/java/util/Optional.html#stream-- and https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#or(java.util.function.Supplier) require Java 9.